src/main/webapp/js/datahandlers/scholarshipdatahandler.js

```diff
@@ -0,0 +1,124 @@
+    requirementsMap.set('Academic Requirements', 
+                        data['academicRequirements'].join(SEPARATOR) || 'N/A');
+    requirementsMap.set('Ethnicity Race Requirements', 
+                        data['ethnicityRaceRequirements'].join(SEPARATOR) || 'N/A');
+    requirementsMap.set('Financial Requirements', 
+                        data['financialRequirements'].join(SEPARATOR) || 'N/A');
+    requirementsMap.set('Gender Requirements', 
+                        data['genderRequirements'].join(SEPARATOR) || 'N/A');
+    requirementsMap.set('Location Requirements', 
+                        data['locationRequirements'].join(SEPARATOR) || 'N/A');
+    requirementsMap.set('National Origin Requirements', 
+                        data['nationalOriginRequirements'].join(SEPARATOR) || 'N/A');
+    requirementsMap.set('Other Requirements', 
+                        data['otherRequirements'].join(SEPARATOR) || 'N/A');
+    return {
+      generalInfo: {
+        scholarshipName: data['scholarshipName'], 
+        scholarshipUUID: data['scholarshipUUID'], 
+        schoolsList: data['schoolsList'].join(SEPARATOR),
+        introduction: data['introduction']['value'] || 'N/A', 
+        URL: data['URL'],
+      },
+      requirements: requirementsMap,
+      applicationNotes: {
+        amountPerYear: data['amountPerYear'] || 'unknown',
+        applicationProcess: data['applicationProcess'] || 'unknown',
+        isRenewable: data['isRenewable'] || 'unknown',
+        numberOfYears: data['numberOfYears'],
+      }, 
+    };
+  };
+
+  /**
+   * Fetch the scholarship data with the specified uuid and format it.
+   * @param {string} id The uuid of the scholarship data.
+   * @return The formatted scholarship JS object map.
+   */
+  async fetchAndFormatSingleScholarshipData(id) {
+    let data = undefined;
+    try {
+      data = await this.fetchScholarshipJson_(id);
+    } catch (e) {
+      console.log(e);
+      throw(`Failed to fetch scholarship object ${e}`);
+    }
+    
+    // If data is undefined, bring user to an error page.
+    // This case will be handled in ScholarshipPageView.
+    // Later after we support querying by uuid, data will be a acholarship object.
+    if (data === undefined) {
+      // This error will be handled and logged by the caller.
+      throw new Error('Cannot get data from remote.');
+    }
+    //In actual project here the data will just be one scholarship object.
+    return this.convertFromJsonToTemplate_(data[id]);
+  }
+
+  /**
+   * Fetch request to the data servlet and return the JSON response.
+   * @param {string} id The uuid of the schedule.
+   * @return {*} - The JSON response.
+   * @private
+   */
+  async fetchScholarshipJson_(id) {
+    const response = await fetch(this.scholarshipEndPoint_, {'id': id });
+    let data = undefined;
+    if (response.ok) {
+      try {
+        data = await response.json();
+        return data;
+      } catch (e) {
+        console.log(e);
```

Comment:
@MikolJiron String formatted error looks nicer in alert panel, but in that way we can't see the stack trace. For now I'll just let the console print the entire error.